transport/http/handler.go

```diff
@@ -109,9 +115,36 @@ func (h handler) callHandler(w http.ResponseWriter, req *http.Request) error {
 			treq.Procedure, deadline.Sub(start))
 	}
 
+	if err != nil {
+		span.SetTag("error", true)
+		span.LogEvent(err.Error())
+	}
+
 	return err
 }
 
+func (h handler) createSpan(ctx context.Context, req *http.Request, treq *transport.Request, start time.Time) (context.Context, opentracing.Span) {
+	// Extract opentracing etc baggage from headers
+	// Annotate the inbound context with a trace span
+	tracer := h.Deps.Tracer()
+	carrier := opentracing.HTTPHeadersCarrier(req.Header)
+	parentSpanCtx, _ := tracer.Extract(opentracing.HTTPHeaders, carrier)
```

Comment:
Errors are being ignored here -- should we be checking them?

If there's an error, the `parentSpanCtx` could be `nil` which we should probably handle